bin/cmd/build

```diff
@@ -162,6 +162,18 @@ if (ghout) {
 ///////////////////////////////////////////////////////////////////
 function make_toolchain() {
   const deps = new Set(config.deps.dry.build.concat(config.deps.dry.runtime).map(x => x.project))
+
+  // gnu.org/binutils ar and ranlib seem to seriously mess up darwin builds in
+  // many cases; if xcode ar/ranlib are available, we'll use them
+  if (deps.has('gnu.org/binutils') &&
+      host().platform == "darwin" &&
+      new Path('/usr/bin/ar').exists() &&
+      new Path('/usr/bin/ranlib').exists()) {
+        const d = config.path.home.join('toolchain').mkdir('p')
+        d.join('ar').ln('s', { target: '/usr/bin/ar' })
+        d.join('ranlib').ln('s', { target: '/usr/bin/ranlib' })
+  }
```

Comment:
I think instead we should add shims for these tools that delegate on darwin and use gnu on linux. Asking for gnu binutils for a build is fine, it's just you would also want to specify gcc most likely as well.

Seems like on linux gnu binutils and llvm work then?

Reply:
as far as i can tell, yes.

Reply:
I can set up the shims

Reply:
we already have shims for these! k so is the problem binutils is needed for some other tool so it must remain on darwin?

Reply:
The problem is that if gcc is used it brings in binutils, which provides these, but they definitively break some builds. There are a few packages that had to explicitly override them via env.

Reply:
kk I get it. It's because our shims for `ar` etc. are removed when the pkg specifies gcc or llvm explicitly.

Which is probs the right call lol. Need more though

Reply:
I don't even see how your fix works. We don’t include the `toolchain` directory in `PATH` if the pkg depends on gcc. So how come you adding symilnks there does anything?

Reply:
Urk. Sounds like I missed a step.

Reply:
well you say it fixed them and I believe you, I just don't see how lol

Reply:
sorry; replacing binutils ar and ranlib via ENV vars fixed the builds in the past. this (obviously incomplete) "fix" was intended to allow removal of that hack. just need to add `toolchain` to the PATH iff it is populated.